controllers/issuer_controller.go

```diff
@@ -171,64 +197,43 @@ func (r *IssuerReconciler) reconcileStatusPatch(
-				&issuerStatusPatch.Conditions,
-				issuer.GetGeneration(),
-				cmapi.IssuerConditionReady,
-				cmmeta.ConditionFalse,
-				v1alpha1.IssuerConditionReasonPending,
-				fmt.Sprintf("Issuer is not ready yet: %s", err),
-			)
-			r.EventRecorder.Eventf(issuer, corev1.EventTypeWarning, "RetryableError", "Failed to check issuer, will retry: %s", err)
-			// We trigger a reconciliation here. Controller-runtime will use exponential backoff to requeue
-			// the request. We don't return an error here because we don't want controller-runtime to log an
-			// additional error message and we want the metrics to show a requeue instead of an error to be
-			// consistent with the other cases (see Permanent error above).
-			//
-			// Important: This means that the ReconcileErrors metric will only be incremented in case of a
-			// apiserver failure (see "unexpected get error" above). The ReconcileTotal labelRequeue metric
-			// can be used instead to get some estimate of the number of requeues.
-			result.Requeue = true
-			return result, issuerStatusPatch, nil // apply patch, retry
-		}
+	if err == nil {
+		logger.V(1).Info("Successfully finished the reconciliation.")
+		message := setCondition(
+			cmapi.IssuerConditionReady,
+			cmmeta.ConditionTrue,
+			v1alpha1.IssuerConditionReasonChecked,
+			"Succeeded checking the issuer",
+		)
+		r.EventRecorder.Event(issuer, corev1.EventTypeNormal, eventIssuerChecked, message)
+
+		return result, issuerStatusPatch, nil // apply patch, done
 	}
 
-	conditions.SetIssuerStatusCondition(
-		r.Clock,
-		issuer.GetStatus().Conditions,
-		&issuerStatusPatch.Conditions,
-		issuer.GetGeneration(),
-		cmapi.IssuerConditionReady,
-		cmmeta.ConditionTrue,
-		v1alpha1.IssuerConditionReasonChecked,
-		"checked",
-	)
-
-	logger.V(1).Info("Successfully finished the reconciliation.")
-	r.EventRecorder.Eventf(issuer, corev1.EventTypeNormal, "Checked", "Succeeded checking the issuer")
-	return result, issuerStatusPatch, nil // done, apply patch
+	isPermanentError := errors.As(err, &signer.PermanentError{})
+	if isPermanentError {
+		// fail permanently
+		logger.V(1).Error(err, "Permanent Issuer error. Marking as failed.")
+		message := setCondition(
+			cmapi.IssuerConditionReady,
+			cmmeta.ConditionFalse,
+			v1alpha1.IssuerConditionReasonFailed,
+			fmt.Sprintf("Issuer has failed permanently: %s", err),
+		)
+		r.EventRecorder.Event(issuer, corev1.EventTypeWarning, eventIssuerPermanentError, message)
+		return result, issuerStatusPatch, reconcile.TerminalError(err) // apply patch, done
+	} else {
+		// retry
+		logger.V(1).Error(err, "Retryable Issuer error.")
+		message := setCondition(
+			cmapi.IssuerConditionReady,
+			cmmeta.ConditionFalse,
+			v1alpha1.IssuerConditionReasonPending,
+			fmt.Sprintf("Issuer is not ready yet: %s", err),
+		)
+		r.EventRecorder.Event(issuer, corev1.EventTypeWarning, eventIssuerRetryableError, message)
+		return result, issuerStatusPatch, err // apply patch, requeue with backoff
```

Comment:
We can now directly return the error to trigger a reconciliation, because the other path returns an error too (using `reconcile.TerminalError`).